markdown_it/tree.py

```diff
@@ -76,6 +83,11 @@ def recursive_collect_tokens(
         recursive_collect_tokens(self, tokens)
         return tokens
 
+    @property
+    def is_root(self) -> bool:
```

Comment:
We could use `self.is_root` inside `def type()` definition to remove duplication.